drivers/dsp0401/Kconfig

```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2020 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+config MODULE_DSP0401
+    bool "DSP0401 alphanumeric display"
+    depends on HAS_PERIPH_GPIO
+    depends on HAS_PERIPH_PWM
+    depends on TEST_KCONFIG
+    select MODULE_PERIPH_GPIO
+    select MODULE_PERIPH_PWM
+    select MODULE_XTIMER
```

Comment:
Why the choice to use select for these? Why not use select in the `feetch` driver for `MODULE_UART_HALF_DUPLEX`?

Reply:
For the case of the peripherals it is safe to use `select` given that their dependency is known (the correspondent `HAS_` feature symbol), which is added as a dependency as well to the driver symbol. That way there is no way of selecting a peripheral driver that is not supported.

In the case of Xtimer I chose to make a trade-off given the massive usage of this module. This decision can of course be reviewed, but I found that an user would have to enable Xtimer for so many modules that it would become to much of a hassle. Of course this choice means that we need to keep a close eye if the dependencies of Xtimer happen to change.

Long chains of `select` are not recommended and should be avoided when possible. As `MODULE_UART_HALF_DUPLEX` has dependencies on itself, using a select would be possible if all those dependencies are also moved up to `MODULE_FEETECH`.

Reply:
Ok makes sense. Let see how this keeps scaling and evolving. I'm curious to see what will result with GNRC.